SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -285,7 +315,39 @@ void ASpatialFunctionalTest::FinishTest(EFunctionalTestResult TestResult, const
 			CurrentStepIndex = SPATIAL_FUNCTIONAL_TEST_FINISHED;
 			OnReplicated_CurrentStepIndex(); // need to call it in Authority manually
 
-			Super::FinishTest(TestResult, Message);
+			CachedTestResult = TestResult;
+			CachedTestMessage = Message;
+
+			GetWorld()->GetTimerManager().SetTimer(
+				FinishTestTimerHandle,
+				[this]() {
+					// If this timer trigger, then it means that something went wrong with one of the Workers. The
+					// expected behaviour is that the Super::FinishTest will be called from Tick(). Let's double check
+					// which failed to reply.
+					FString WorkersDidntAck;
+					for (const auto* FlowController : FlowControllers)
+					{
+						if (!FlowController->HasAckFinishedTest())
+						{
+							WorkersDidntAck += FString::Printf(TEXT("%s, "), *(FlowController->GetDisplayName()));
+						}
+					}
+					if (!WorkersDidntAck.IsEmpty())
+					{
+						WorkersDidntAck.RemoveFromEnd(TEXT(", "));
+						UE_LOG(LogSpatialGDKFunctionalTests, Warning,
+							   TEXT("The following Workers failed to acknowledge FinishTest in time: %s"), *WorkersDidntAck);
+					}
+
+					Super::FinishTest(CachedTestResult, CachedTestMessage);
+
+					FinishTestTimerHandle.Invalidate();
+
+					// Clear cached values.
+					CachedTestResult = EFunctionalTestResult::Default;
+					CachedTestMessage.Empty();
+				},
+				2.0f /* InRate */, false /* InbLoop */);
```

Comment:
maybe a constant somewhere if we go ahead with this design

Reply:
I can add a constant at the top.